operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -158,26 +185,50 @@ public static File createFileStore(String prefix, String suffix, byte[] bytes) {
      * Decode binary item from Kubernetes Secret from base64 into byte array
      *
      * @param secret    Kubernetes Secret
-     * @param key       Key which should be retrieved and decoded
+     * @param field     Field which should be retrieved and decoded
      * @return          Decoded bytes
      */
-    public static byte[] decodeFromSecret(Secret secret, String key) {
-        return Base64.getDecoder().decode(secret.getData().get(key));
+    public static byte[] decodeBase64FieldFromSecret(Secret secret, String field) {
+        return Optional.ofNullable(secret)
+                .map(Secret::getData)
+                .map(data -> data.get(field))
+                .map(value -> Base64.getDecoder().decode(value))
+                .orElseThrow(() -> {
+                    String name = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getName)
+                            .orElse("unknown");
+                    String namespace = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getNamespace)
+                            .orElse("unknown");
```

Comment:
When would it not have a namespace?

Reply:
I guess it would be better to check that separately and possibly throw a different error and not mask it behind some `unknown` namespace?